src/BaseVault.sol

```diff
@@ -332,6 +332,11 @@ abstract contract BaseVault is IVault, ERC20PermitUpgradeable, AccessControlUpgr
         }
         shares = previewWithdraw(assets);
 
+        IHooks hooks_ = hooks();
+        if (address(hooks_) != address(0)) {
```

Comment:
i'm wondering if for cleanliness we mandate a hooks implementation so if there isn't one set it reverts. and there's a NOOP hook to attach.

the address(0) check is a toggle that helps i guess to reduce gas usage.

I do notice uni v4 doesn't check for address Zero, and they usually care about gas usage more than YieldNest as a DEX with very frequent swaps.